packages/mobile/src/transactions/saga.ts

```diff
@@ -98,6 +112,53 @@ export function* sendAndMonitorTransaction<T>(
   }
 }
 
+export function* refreshRecentTxRecipients() {
+  const addressToE164Number = yield select(addressToE164NumberSelector)
+  const recipientCache = yield select(recipientCacheSelector)
+  const knownFeedTransactions: KnownFeedTransactionsType = yield select(
+    knownFeedTransactionsSelector
+  )
+
+  // No way to match addresses to recipients without caches
+  if (
+    !Object.keys(recipientCache).length ||
+    !Object.keys(addressToE164Number).length ||
+    !Object.keys(knownFeedTransactions).length
+  ) {
+    return
+  }
+
+  const knownFeedAddresses = Object.values(knownFeedTransactions)
+
+  let remainingCacheStorage = RECENT_TX_RECIPIENT_CACHE_LIMIT
+  const recentTxRecipientsCache: NumberToRecipient = {}
+  // Start from back of the array to get the most recent transactions
```

Comment:
I don't know that object.values has any guarantees about ordering `The ordering of the properties is the same as that given by looping over the property values of the object manually` https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values

I think if you expect them to be ordered by timestamp it's worth making sure somehow

Reply:
I originally tried creating a Map for that reason but don't think we can store those in Redux.

Given the logic is not visible to the user (i.e., no user expectation of having any recipients cached) and in practice [order is generally maintained for consistent property types](https://www.stefanjudis.com/today-i-learned/property-order-is-predictable-in-javascript-objects-since-es2015/), I opted the complexity of doing a timestamp comparison